spec/02-integration/09-hybrid_mode/12-errors_spec.lua

```diff
@@ -0,0 +1,285 @@
+local helpers = require "spec.helpers"
+local cjson = require "cjson.safe"
+local uuid = require "kong.tools.uuid"
+local constants = require "kong.constants"
+
+local CONFIG_PARSE = constants.CLUSTERING_DATA_PLANE_ERROR.CONFIG_PARSE
+local RELOAD = constants.CLUSTERING_DATA_PLANE_ERROR.RELOAD
+
+local CLUSTER_PORT = helpers.get_available_port()
+local CLUSTER_SSL_PORT = helpers.get_available_port()
+
+
+local function json(data)
+  return {
+    headers = {
+      ["accept"] = "application/json",
+      ["content-type"] = "application/json",
+    },
+    body = assert(cjson.encode(data)),
+  }
+end
+
+
+local mock_cp = ([[
+  lua_shared_dict kong_test_cp_mock 10m;
+
+  server {
+      charset UTF-8;
+      server_name kong_cluster_listener;
+      listen %s;
+      listen %s ssl;
+
+      access_log ${{ADMIN_ACCESS_LOG}};
+      error_log  ${{ADMIN_ERROR_LOG}} ${{LOG_LEVEL}};
+
+> if cluster_mtls == "shared" then
+      ssl_verify_client   optional_no_ca;
+> else
+      ssl_verify_client   on;
+      ssl_client_certificate ${{CLUSTER_CA_CERT}};
+      ssl_verify_depth     4;
+> end
+      ssl_certificate     ${{CLUSTER_CERT}};
+      ssl_certificate_key ${{CLUSTER_CERT_KEY}};
+      ssl_session_cache   shared:ClusterSSL:10m;
+
+      location = /v1/outlet {
+          content_by_lua_block {
+              require("spec.fixtures.mock_cp").outlet()
+          }
+      }
+
+      location = /payload {
+          content_by_lua_block {
+              require("spec.fixtures.mock_cp").set_payload()
+          }
+      }
+
+      location = /log {
+          content_by_lua_block {
+              require("spec.fixtures.mock_cp").get_log()
+          }
+      }
+  }
+]]):format(CLUSTER_PORT, CLUSTER_SSL_PORT)
```

Comment:
Not a blocker for merging this PR, but this could be refactored as a function and added to helpers.

Reply:
Good call! All of this is tightly-coupled to the `spec.fixtures.mock_cp` module, so I moved the template over there.